lua/ttt2/libraries/targetid.lua

```diff
@@ -0,0 +1,558 @@
+---
+-- @module TargetID
+-- @desc Handling operations to display the TargetID
+-- @author Mineotopia
+
+if SERVER then
+    AddCSLuaFile()
+    return
+end
+
+targetid = targetid or {}
+
+-- Global to local variables
+local bIsInitialized = false
+local ParT
+local TryT
+
+-- Variables for calculations
+local MAX_TRACE_LENGTH = math.sqrt(3) * 32768
+
+-- Key Parameters for doors
+local key_params = {}
+
+-- Convars for targetid
+ local cvDeteOnlyConfirm = GetConVar("ttt2_confirm_detective_only")
+ local cvDeteOnlyInspect = GetConVar("ttt2_inspect_detective_only")
+
+-- Materials for targetid
+local materialTButton = Material("vgui/ttt/tid/tid_big_tbutton_pointer")
+local materialRing = Material("effects/select_ring")
+local materialRoleUnknown = Material("vgui/ttt/tid/tid_big_role_not_known")
+local materialDisguised = Material("vgui/ttt/perks/hud_disguiser.png")
+local materialCorpse = Material("vgui/ttt/tid/tid_big_corpse")
+local materialCredits = Material("vgui/ttt/tid/tid_credits")
+local materialDetective = Material("vgui/ttt/tid/tid_detective")
+local materialLocked = Material("vgui/ttt/tid/tid_locked")
+local materialAutoClose = Material("vgui/ttt/tid/tid_auto_close")
+local materialDoor = Material("vgui/ttt/tid/tid_big_door")
+local materialDestructible = Material("vgui/ttt/tid/tid_destructible")
+local materialDNATargetID = Material("vgui/ttt/dnascanner/dna_hud")
+
+---
+-- This function is a temporary fix for local variables in use of other libraries, that are not yet initialized
+function targetid.initialize()
+    if not bIsInitialized then
+        bIsInitialized = true
+        ParT = LANG.GetParamTranslation
+        TryT = LANG.TryTranslation
+        key_params = {
+            usekey = Key("+use", "USE"),
+            walkkey = Key("+walk", "WALK")
+        }
+    end
+end
```

Comment:
```suggestion
function targetid.Initialize()
    if bIsInitialized then return end

    bIsInitialized = true
    ParT = LANG.GetParamTranslation
    TryT = LANG.TryTranslation
    key_params = {
        usekey = Key("+use", "USE"),
        walkkey = Key("+walk", "WALK")
    }
end
```

Reply:
Also I'm pretty sure that those Keyparams can be used without this initialization

Reply:
well maybe I failed the test, but for some reason it did not work, when having only them outside

Reply:
![grafik](https://user-images.githubusercontent.com/72046466/97197629-29ebce00-17ae-11eb-87f6-847b143a5d36.png)
That's what I get if I put the key_params outside.

Reply:
huh, interesting. I thought this was a gmod function